crypto/crypto_aesctr.c

```diff
@@ -135,6 +135,24 @@ crypto_aesctr_stream(struct crypto_aesctr * stream, const uint8_t * inbuf,
     uint8_t * outbuf, size_t buflen)
 {
 	size_t bytemod;
+	size_t nbytes;
+
+	/* Do we have any bytes left in the current cipherblock? */
+	bytemod = stream->bytectr % 16;
+	if (bytemod != 0) {
+		/* How many bytes can we process? */
+		nbytes = 16 - bytemod;
+		if (nbytes > buflen) {
```

Comment:
This could be `>=`, and would probably speed up the small buffer case very slightly.

That said, I would probably write `if (bytemod + buflen <= 16) { ... }` and omit `nbytes` entirely.

Reply:
Yes, that works better, and avoids the pesky `size_t nbytes` that I disliked.